Competition2018/src/main/java/competition/subsystems/climb/ClimbSubsystem.java

```diff
@@ -1,38 +1,60 @@
 package competition.subsystems.climb;
 
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.LimitSwitchNormal;
+import com.ctre.phoenix.motorcontrol.LimitSwitchSource;
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 
 import competition.ElectricalContract2018;
+import competition.subsystems.climberdeploy.ClimberDeploySubsystem;
 import xbot.common.command.BaseSubsystem;
 import xbot.common.controls.actuators.XCANTalon;
 import xbot.common.controls.actuators.XSolenoid;
 import xbot.common.injection.wpi_factories.CommonLibFactory;
+import xbot.common.math.MathUtils;
 import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.StringProperty;
 import xbot.common.properties.XPropertyManager;
 
 @Singleton
 public class ClimbSubsystem extends BaseSubsystem {
 
+    public enum ClimbRestrictionReason {
+        TooMuchStrapOut,
+        TooCloseToWinch,
+        AheadOfDeployArm,
+        FullPowerAvailable
+    }
```

Comment:
This has been really helpful when debugging. We should consider this as some baseline thing for all systems that restrict power in some way.